reps/2022-12-08-ray-for-federated-learning-and-privacy-preserving-computing.md

```diff
@@ -0,0 +1,167 @@
+
+## Summary - Ray for federated learning and privacy-preserving computing
+### General Motivation
+Federated machine learning and privacy-preserving computing are getting wider. They're obviously distributed systems because they're always across parties(companies or ORGs.) To extend Ray ecosystem to federated learning and privacy-preserving computing domain, we should let users be easy to use Ray, for building their own federated learning and privacy-preserving computing applications without any concern on data privacy, task-attacks, illegal invasion, and etc.
+
+PS: I also found another related issue: https://github.com/ray-project/ray/issues/25846
+
+In this proposal, we'd like to build a connector layer on Ray, to provide the ability for users use Ray easily to build their such kind of system, with the secure issues addressed totally.
+
+### Key requirements:
+- Provide the ability for users to easily build their such kind of application on Ray.
+- Setup different Ray clusters for different parties to avoid uncotrollable complex Ray protocols.
+- Have a unified and global-viewed programming mode for different parties.
+- Data transmition across parties should be in push mode instead of pull mode.
+- Tasks should be driven in multi-controller mode. That means tasks should be driven by themselves(who are inside this party) instead of others.
+
+### Should this change be within ray or outside?
+Yes.
+1. A `ray/python/ray/fed` directory should be added to ray repo as the connector layer, like ray collective, ray dag.
```

Comment:
For not introducing unnecessary heavy dependencies, I believe it's better to create a new repository instead of introducing it into `ray` repo.

So the current section should be:
```md
### Should this change be within ray or outside?
No
1. Adding a new repo `RayFed` under `ray-project`: `ray-project/RayFed`
2. The package name is `rayfed`: pip install rayfed
3. importing lines should be `import fed`
```

Reply:
OK, it can be an independent repo for now.